docs/userguide/networking/dockernetworks.md

```diff
@@ -0,0 +1,368 @@
+
+To create an overlay network, you configure options on  the `daemon` on each Docker Engine for use with `overlay` network. There are two options to set:
+
+| Option                           | Description                                               |
+|----------------------------------|-----------------------------------------------------------|
+| `--cluster-store=PROVIDER://URL` | Describes the location of the KV service.                 |
+| `--cluster-advertise=HOST_IP`    | Advertises containers created by the HOST on the network. |
+
+Create an `overlay` network on one of the machines in the Swarm.
+
+        $ docker network create -d overlay my-multi-host-network
+
+This results in a single network spanning multiple hosts. The `overlay` network provides complete isolation for the containers.
+
+![](images/overlay_network.png)
+
+Then, on each host, launch containers making sure to specify the network name.
+
+        $ docker run run -itd --net=mmy-multi-host-network busybox
+
+Once launched, each container has access to all the containers in the network regardless of which Docker host the container was launched on.
+
+![](images/overlay-network-final.png)
+
+If you would like to try this for yourself, see the [Getting started for overlay](get-started-overlay.md).
+
+### Custom network plugin
+
+If you like, you can create your own custom, network driver plugin. A network
+driver plugin makes use of Docker's plugin infrastructure. In this
+infrastructure, a plugin is a process running on the same Docker host as the
+Docker `daemon`.
+
+Network plugins follow the same restrictions and installation rules as other
+plugins. All plugins make use of the plugin API. They have a lifecycle that encompasses installation, starting, stopping and activation.
+
+Once you have creatd a custom network driver, you use it like the built-in
+network drivers. For example,
+
+    $ docker network create -d weave mynet
+
+You can inspect it, add containers too and from it, and so forth. Of course, different plugins may make use of different technologies or frameworks so can add features not present in Docker's default networks.  For more information on writing plugins, see (Extending Docker)[../../extend] and (Writing a network driver plugin)[../../extend/plugins_network.md].
+
+## Legacy links
+
+Before the Docker network feature, you could use the Docker link feature to allow containers to discover each other and securely transfer information
+about one container to another container. With the introduction of Docker networks, you can still create links but they are only supported on the default `bridge` network named `bridge` and appearing in your network stack as `docker0`.
+
+While links are still supported in this limited capacity, you should avoid them in preference of Docker networks. The link feature is expected to be deprecated and removed in a future release.
```

Comment:
Should we mention possible future deprecations here? (I know there's still discussions ongoing around that)